aten/src/ATen/native/metal/MetalPrepackOpRegister.cpp

```diff
@@ -102,6 +102,99 @@ Tensor conv2d_prepack_run(
 #endif
 }
 
+
+c10::intrusive_ptr<Conv3dOpContext> unpack(
+    Tensor&& weight,
+    c10::optional<Tensor>&& bias,
+    std::vector<int64_t>&& stride,
+    std::vector<int64_t>&& padding,
+    std::vector<int64_t>&& dilation,
+    int64_t groups,
+    c10::optional<Scalar> output_min,
+    c10::optional<Scalar> output_max) {
+  const Tensor weightContig = weight.contiguous();
+  const auto ws = weightContig.sizes();
+  auto packed_buffer = permuteWeights(weightContig.data_ptr<float>(), ws.vec());
+  auto packedWeight = at::empty(ws);
+  int64_t size_bytes = at::prod_intlist(ws) * sizeof(float);
+  memcpy(packedWeight.data_ptr(), packed_buffer.data(), size_bytes);
+  return c10::make_intrusive<Conv3dOpContext>(
+      std::move(packedWeight),
+      std::move(bias),
+      stride,
+      padding,
+      dilation,
+      groups,
+      output_min,
+      output_max);
+}
+
+TORCH_LIBRARY(metal, m) {
```

Comment:
redefinition of 'TORCH_LIBRARY_init_metal'

Reply:
I wasn't confident with the cpp syntax. I can rename those functions so that the "_2D" /"_3D" suffix appears in the name, but I'm not sure where I need to propagate this change. Do you know what I need to look at to do that ?

Reply:
Resolving this because we started the same convo in another thread.